channeldb/db.go

```diff
@@ -226,7 +226,7 @@ func (db *DB) Update(f func(tx walletdb.ReadWriteTx) error, reset func()) error
 	}
 
 	reset()
-	return walletdb.Update(db, f)
+	return walletdb.Update(db.Backend, f)
```

Comment:
nit: Could be squashed with the previous commit.

Reply:
I wanted to give this special attention in case it has any side effects I wasn't aware of, that's why it is in its own commit.

Reply:
SGTM

Reply:
Was able to remove this commit after the latest set of updates (following your idea).